tools/pkg/engine_build_configs/lib/src/ci_yaml.dart

```diff
@@ -0,0 +1,221 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:yaml/yaml.dart' as y;
```

Comment:
For a few months, I maintained package:cocoon_scheduler which contained Dart helpers for stuff like this used by Cocoon. I ended up removing it in https://github.com/flutter/cocoon/pull/1254 as Cocoon ended up being the only customer, and it was annoying to roll the versions.

For simplicity, would you be blocked from importing the Cocoon server into the engine? We could set up a script to publish new versions to pub.dev. If it pulls in too much, we could move https://github.com/flutter/cocoon/tree/main/app_dart/lib/src/model/ci_yaml into a separate package.

\cc @keyonghan

Reply:
> It depends on how extensive the transitive dependencies will be, since they would all need to be explicitly brought in via the DEPS file. If it's just a few things, then that's fine, but if the cocoon scheduler has a big list of transitive deps, then that would be a lot of trouble.

Is there a way we can estimate that by importing it in pubspec via a [git package](https://dart.dev/tools/pub/dependencies#git-packages)?

> @CaseyHillers I'm having trouble finding where in https://github.com/flutter/cocoon/tree/main/app_dart/lib/src/model/ci_yaml the yaml file is parsed. It looks like the code in that directory is mostly poking around in the fields of a protobuf.

ci.yaml is proto backed, so the pipeline looks like package:yaml -> toJson() -> package:protobuf mergeFromProtoJson -> ci_yaml.dart

https://github.com/flutter/cocoon/blob/main/app_dart/lib/src/service/scheduler.dart#L272

For Cocoon, it reads the file content over the network, then 

```dart
    final String configContent = await githubFileContent(
      commit.slug,
      '.ci.yaml',
      ref: commit.sha!,
    );
    final YamlMap configYaml = loadYaml(configContent) as YamlMap;
    final pb.SchedulerConfig schedulerConfig = pb.SchedulerConfig()
      ..mergeFromProto3Json(configYaml)
      ..writeToBuffer();
```